akka-http/src/main/scala/akka/http/scaladsl/common/NameReceptacle.scala

```diff
@@ -16,19 +17,35 @@ class NameReceptacle[T](val name: String) {
   def as[B] = new NameReceptacle[B](name)
   def as[B](unmarshaller: Unmarshaller[T, B])(implicit fsu: FSU[T]) =
     new NameUnmarshallerReceptacle(name, fsu.transform[B](implicit ec => implicit mat => { _ flatMap unmarshaller.apply }))
+  /** Symbolic alias for [[optional]]. */
   def ? = new NameOptionReceptacle[T](name)
+  def optional = new NameOptionReceptacle[T](name)
+  /** Symbolic alias for [[withDefault]]. */
   def ?[B](default: B) = new NameDefaultReceptacle(name, default)
+  def withDefault[B](default: B) = new NameDefaultReceptacle(name, default)
+  /** Symbolic alias for [[requiredValue]]. */
   def ![B](requiredValue: B) = new RequiredValueReceptacle(name, requiredValue)
+  def requiredValue[B](requiredValue: B) = new RequiredValueReceptacle(name, requiredValue)
+  /** Symbolic alias for [[repeated]]. */
   def * = new RepeatedValueReceptacle[T](name)
+  def repeated = new RepeatedValueReceptacle[T](name)
 }
 
 class NameUnmarshallerReceptacle[T](val name: String, val um: FSU[T]) {
   def as[B](implicit unmarshaller: Unmarshaller[T, B]) =
     new NameUnmarshallerReceptacle(name, um.transform[B](implicit ec => implicit mat => { _ flatMap unmarshaller.apply }))
+  /** Symbolic alias for [[optional]]. */
   def ? = new NameOptionUnmarshallerReceptacle[T](name, um)
+  def optional = new NameOptionUnmarshallerReceptacle[T](name, um)
+  /** Symbolic alias for [[withDefault]]. */
   def ?(default: T) = new NameDefaultUnmarshallerReceptacle(name, default, um)
+  def withDefault(default: T) = new NameDefaultUnmarshallerReceptacle(name, default, um)
```

Comment:
Good to have names as well here.

I find the naming a bit inconsistent with `optional`, `withDefault`, and `requiredValue` being different grammatical forms but if we have the same names on the Java side, let's keep them for consistency with those. If we have better ideas, we can still introduce different aliases later on.